Fix intermittent failure in RunRoleManagementServiceBlackboxTest

We are still experiencing failures of this test on some occasions:

```
--- FAIL: TestRunRoleManagementServiceBlackboxTest (0.63s)
    --- FAIL: TestRunRoleManagementServiceBlackboxTest/TestGetRolesByResourceTypeOK (0.02s)
        assertions.go:247: 
                          
    Error Trace:    role_management_service_blackbox_test.go:174
            
                    
            role_management_service_blackbox_test.go:119
            
    Error:          Received unexpected error:
            
                    role with id '67f9e88c-39ee-42e0-9b3c-cf706f8fd1dc' not found
            
    Test:           TestRunRoleManagementServiceBlackboxTest/TestGetRolesByResourceTypeOK
```

--- a/account/identity_blackbox_test.go
+++ b/account/identity_blackbox_test.go
@@ -163,7 +163,7 @@ func (s *identityBlackBoxTest) TestFindIdentityMemberships() {
 	// Create an identity
 	identity := createAndLoad(s)
 
-	orgName := "Acme Corporation - identityBlackBoxTest.TestFindIdentityMemberships"
+	orgName := "Acme Corporation - identityBlackBoxTest.TestFindIdentityMemberships" + uuid.NewV4().String()
 	// Create an organization
 	orgID, err := s.Application.OrganizationService().CreateOrganization(s.Ctx, identity.ID, orgName)
 	require.NoError(s.T(), err)

--- a/authorization/role/repository/role.go
+++ b/authorization/role/repository/role.go
@@ -270,7 +270,9 @@ func (m *GormRoleRepository) FindRolesByResourceType(ctx context.Context, resour
 			resource_type rt
 		WHERE  
 			rt.resource_type_id = r.resource_type_id 
+      AND r.deleted_at IS NULL
 			AND rt.NAME = ?
+      AND rt.deleted_at IS NULL
 		GROUP BY 
 		  r.role_id, 
 		  r.name`, resourceType)

